services/bulletin/src/main/java/edu/uga/devdogs/bulletin/webscraping/BulletinScraper.java

```diff
@@ -0,0 +1,197 @@
+package edu.uga.devdogs.bulletin.webscraping;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class BulletinScraper {
+    // This method takes a line from the table and attempts to extract a course ID from it
+    private static String getCourseIDFromTable(String text) {
+        // Table entry will look something like "ABCD 1234." Look at the text until we reach the period.
+        return text.substring(0,text.indexOf("."));
+    }
+
+    // This method will take the line from the table that contains the prerequisites and parse it
+   private static ArrayList<PrerequisiteGroup> getPrerequisiteGroupFromTable(String text, String courseName) {
```

Comment:
javadoc for all methods
ex:
/**
* short descripton of what the method does
* @param text - what you are expecting for the text input, if applicable an example of sample input
* @param courseName. - 
* @return what you are returning
* /